.changeset/early-lions-turn.md

```diff
@@ -0,0 +1,21 @@
+---
+'@leafygreen-ui/code': major
+---
+
+Fixes various bugs, makes accessibility improvements, tweaks the design, and improves maintainability of the `Code` component.
+
+Design changes:
+- Updates the syntax highlighting theme to improve contrast between comments and regular code, as well as making several optimizations for red-green colorblindness.
+- Tightens the top and bottom padding for the code component, especially for single line code examples.
+- Significantly tightens the left padding when line numbers are shown.
+- Right-aligns line numbers in their column to improve readability.
+
+Bug fixes:
+- Fixes an issue where iOS browsers would render code examples at a `20px` size, rather than the `13px` font size they're supposed to render as.
+- Sometimes code that's written as multi-line statements wouldn't have line numbers applied to each line of the statement. That should now be fixed.
```

Comment:
This is fixed as far as I can tell, do we know what was causing it in the first place? Is there a test that we can add to make sure we don't introduce a regression later?

Reply:
This was technically not a bug. Apparently the specs for browsers to implement are unspecific when it comes to rendering text with `white-space: pre`. iOS specifically applies a font size that can't be overwritten on mobile of `20px`, and unfortunately that means that in those cases, we have to wrap long lines of code instead of overflow scroll handling it to fix this issue. CSS-only fixes are difficult to effectively test without visual diffing, so I don't think there's a good test to add here.

Reply:
Ignore my last response – it was in regard to something else.

highlight.js was outputting a nested structure for the parsed code to highlight with. We weren't doing anything with that nested structure, causing it to be output on the same effective line regardless of a line break. I'll add a test for the function I added to flatten that data structure.

Reply:
sgtm, thank you. just want to make sure we're holding ourselves accountable to this solution